File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        if not nums: return 0
        
        num_set = set(nums)
        result = 0
        
        for num in num_set:
            if num-1 not in num_set:
                current_num = num
                current_max = 1
                
                while current_num + 1 in num_set:
                    current_max += 1
                    current_num += 1
                result = max(result, current_max)
        
        return result

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        if (nums.size() < 2) {
            return nums.size();
        }
        
        sort(nums.begin(), nums.end());
        
        int n = 1;
        int result = n;
        
        for(int i = 0; i < nums.size()-1; i++) {
            if (nums[i+1] == nums[i]) {
                continue;
            }
            if (nums[i+1] - nums[i] == 1) {
                n += 1;
                result = max(result, n);
            }
            else {
                n = 1;
            }
        }
        return result;
        
    }
};